### PR TITLE
Use aws sdk v2 single object delete (not bulk delete) for S3FileIO

### PR DESCRIPTION
**Pull request by danielcweeks**

### CODE REVIEW

File: aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java [hunk 1]
```diff
@@ -88,12 +86,10 @@ public OutputFile newOutputFile(String path) {
   @Override
   public void deleteFile(String path) {
```
Comment (jacques-n) [inline aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java:87]: Is there a guarantee that the path isn't a "directory"?

Comment (danielcweeks) [reply aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java:87]: There's really no concept of a directory (in the sense that you can recursively delete a bunch of data) in S3.  Directories are just a byproduct of whatever you use as a delimiter in listing (most typically `/` to make it look like a filesystem structure).  Some FileSystem implementation use a marker for empty directories (e.g. `<path>_$folder$`), but that's just to workaround the fact that there isn't such a thing.

For Iceberg, this should be safe as listing isn't used and files are all directly referenced.

**Review by jackye1995 — APPROVED** [review]
looks good to me

**Comment by jackye1995** [comment]
btw, there is a spelling mistake at https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputFile.java#L60, could you also fix that when merging this PR?

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks @danielcweeks!

@jackye1995, I didn't wait to fix that typo because I didn't want this commit to touch an additional unrelated file. We can fix that in a separate PR.

**Pull request merged**